core/src/main/java/cucumber/api/Result.java

```diff
@@ -78,11 +78,15 @@ public boolean is(Result.Type status) {
     }
 
     public boolean isOk(boolean isStrict) {
-        return hasAlwaysOkStatus() || !isStrict && hasOkWhenNotStrictStatus();
+        return hasAlwaysOkStatus(false) || !isStrict && hasOkWhenNotStrictStatus();
     }
 
-    private boolean hasAlwaysOkStatus() {
-        return is(Result.Type.PASSED) || is(Result.Type.SKIPPED);
+    public boolean isOkWip(boolean isStrict) {
+        return hasAlwaysOkStatus(true) || !isStrict && hasOkWhenNotStrictStatus();
```

Comment:
I don't think this method is needed. 

When using  `--wip` the build should fail  when there are any passing results (when result.isOk(true/false) return true).

Reply:
@mpkorstanje If I only use isOk, and I do what you tell, I obtain an unexpected behaviour with strict mode.

With wip and strict, if I have an undefined step, the result is 0 instead of 1.

Reply:
I thought it wasn't possible to use wip and strict in together. But if it is, that makes sense. An undefined result in strict mode makes the test fail. So when running with wip and strict  it should pass. That is what I get from reading [work_in_progress.feature](https://github.com/cucumber/cucumber-ruby/blob/d73e337a5d347313984f811e8c9b139e74c06551/features/docs/work_in_progress.feature):

```
  In order to ensure that feature scenarios do not pass until they are expected to
  Developers should be able to run cucumber in a mode that
            - will fail if any scenario passes completely
            - will not fail otherwise
```

Reply:
Ok, do you want I disable the use of strict with wip ?

Reply:
Why do you think the behaviour is unexpected?

Reply:
I mentioned somewhere else that I couldn’t remember why we disallowe strict and wip at the same time.

If we can’t think of a good reason for it (I can’t) we should allow it.

Reply:
I am thinking the same thing. But I am trying to work out why Heziode thinks the result of combining wip and strict with an undefined result is unexpected.

Reply:
Here is a result I want expect:

- `--wip`: Test fail only if one or more scenario have status `PASSED`
- `--wip` + `--strict`: Test fail if one or more scenario have status `PASSED`, `UNDEFINED`, `SKIPPED`

What do you think about this ?